### PR TITLE
WIP: Added option to prevent deletion but still allow uploads

### PR DESCRIPTION
**Pull request by Nepoxx**
I used the `-w` (write only?) for the cli as `-n` and `-d` were used.

### CODE REVIEW

**Comment by derhuerst** [comment]
I'm fine with the intention of this PR, but in this case please change the `--readonly` flag to `allow-upload` and `--nodelete` to `--allow-delete` to make it consistent (the defaults should stay like before).

**Comment by Nepoxx** [comment]
I used `nodelete` to stay consistent with `readonly`.

If we go with `allow-upload`, we can't keep the same defaults since `allow-upload` is on by default.

**Comment by derhuerst** [comment]
> I used `nodelete` to stay consistent with `readonly`.

I'm aware that my proposal is a breaking change. Nevertheless I find it more intuitive to have flags that enable things than flags that disable something.

**Comment by derhuerst** [comment]
Considering that passing `false` as an option value in the command line seems awkward, I'm actually with your proposal. I don't know what I was thinking a few hours ago. 😛 

What about `--no-delete` and `--no-upload`?

**Comment by Nepoxx** [comment]
That sounds good, would you keep `--readonly` which would be a combination of `--no-delete` and `--no-upload`?

**Comment by derhuerst** [comment]
Yup, I would.

**Comment by Nepoxx** [comment]
I'm leaving the country for a week, I'll get this done on my return.

File: lib/index.js [hunk 1]
```diff
@@ -21,7 +21,8 @@ const defaults = {
 	name: 'casket',
 	root: process.cwd(),
 	readonly: false,
-	nodelete: false,
+	'no-delete': false,
```
Comment (derhuerst) [inline lib/index.js]: I'd prefer `noDelete` and `noUpload` here, as this part of the code is not related to the command line interface and should therefore have idiomatic variable names.

Comment (Nepoxx) [reply lib/index.js]: That's a very fair observation, fixed!

**Comment by Nepoxx** [comment]
`-w` and `-n` flags work as expected, `--no-delete` and `--no-upload` do not. I'm working on it.

**Comment by Nepoxx** [comment]
@derhuerst I'm not familiar with mri, it is returning `--no-delete` as `delete` (if I pass `--no-delete` only `argv.delete` is set). I believe this is a normal behaviour of mri (to prevent having to use the `argv['no-delete']` notation), thoughts?

**Comment by derhuerst** [comment]
Yes, [apparently `mir` magically parses `--no-…` flags](https://github.com/lukeed/mri/blob/22df50ca04e2e474765bb4e2ec1bf8a052db9ed9/test/index.js#L7).

File: cli.js [hunk 2]
```diff
@@ -10,17 +10,19 @@ Usage:
...
+    --no-upload -u  Do not allow file upload.
+    --port      -p  Default is 8000.
 
 `
 
 
 
 const argv = mri(process.argv.slice(2), {
-	boolean: ['help', 'h', 'readonly', 'r']
+	boolean: ['help', 'h', 'readonly', 'r', 'no-delete', 'w', 'no-upload', 'u']
```
Comment (derhuerst) [inline cli.js:25]: Not sure if this is necessary anymore. Otherwise every LGTM!

Comment (Nepoxx) [reply cli.js:25]: I don't think it is because of the `=== false`, but I think it makes it more consistent with the other boolean options

**Comment by derhuerst** [comment]
Thanks for your patience!

**Pull request merged**